modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4156,21 +4155,7 @@ def compute_groupby(df, drop=False, partition_idx=0):
         else:
             apply_indices = None
 
-        if (
-            # For now handling only simple cases, where 'by' columns are described by a single query compiler
-            agg_kwargs.get("as_index", True)
-            and len(not_broadcastable_by) == 0
-            and len(broadcastable_by) == 1
-            and broadcastable_by[0].has_materialized_dtypes
-        ):
-            new_index = ModinIndex(
```

Comment:
@dchigarev  How is it supposed to take the length from this object in `broadcast_apply_full_axis` function if it refers to the old modin frame?

Reply:
@dchigarev ready for review